client/src/app/site/agenda/components/list-of-speakers/list-of-speakers.component.ts

```diff
@@ -339,9 +377,26 @@ export class ListOfSpeakersComponent extends BaseViewComponent implements OnInit
      * @param speaker The speaker clicked on.
      */
     public onMarkButton(speaker: ViewSpeaker): void {
-        this.listOfSpeakersRepo
-            .markSpeaker(this.viewListOfSpeakers, speaker, !speaker.marked)
-            .then(null, this.raiseError);
+        this.listOfSpeakersRepo.markSpeaker(this.viewListOfSpeakers, speaker, !speaker.marked).catch(this.raiseError);
+    }
+
+    /**
+     * Removes the last finished speaker from the list an re-adds him on pole position
+     * Re-adding and sorting has to be done seperately right now which causes a quick jump in the list
+     */
+    public async addLastSpeakerToList(): Promise<void> {
+        const lastSpeaker = this.finishedSpeakers.slice(-1)[0];
+        try {
+            await this.listOfSpeakersRepo.delete(this.viewListOfSpeakers, lastSpeaker.id);
+            this.filterUsers();
+        } catch (e) {
+            this.raiseError(e);
+        }
+        this.listOfSpeakersRepo.createSpeaker(this.viewListOfSpeakers, lastSpeaker.user_id).then(newSpeaker => {
```

Comment:
Even if this method will be obsolete, you are mixing a callback-style Promises with async/await syntax. Instead of `then` just awaiting the promise. This leads to unnesting the callbacks and putting all statements into an try-catch you have to do errorhandling just once.